sympy/functions/elementary/tests/test_miscellaneous.py

```diff
@@ -461,12 +465,27 @@ def test_issue_14000():
     assert root(16, 4, 2, evaluate=False).has(Pow) == True
     assert real_root(-8, 3, evaluate=False).has(Pow) == True
 
+
 def test_issue_6899():
     from sympy.core.function import Lambda
     x = Symbol('x')
     eqn = Lambda(x, x)
     assert eqn.func(*eqn.args) == eqn
 
+
+def test_MinMax_with_Sets():
+    assert Max(Range(3)) == 2
```

Comment:
I think this might be conflating ideas of objects and object elements. In a similar way we needed to deistinguish between `FiniteSet(FiniteSet(1, 2)) != FiniteSet(1, 2)`. When the user writes `Max(a, b)` then `a` and `b` must be compared, not the elements within. If there is code that needs to find the largest item in sets then it should be `max(i.sup for i in sets)`.

Reply:
I think `max(Range)` is ok but have doubts about `max(Range, FiniteSet)`. Note how the builtin behaves:
```python
>>> max([1,2,3],[2])
[2]
```
So it returned the element that had the largest first difference. And mixed comparisons fail:
```python
>>> max([1,2,3],[2],7)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '>' not supported between instances of 'int' and 'list'
```